LayoutKitObjCSampleApp/ViewController.m

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 LinkedIn Corp.
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+#import "ViewController.h"
+#import "RotationLayout.h"
+
+@import Foundation;
+@import UIKit;
+
+@interface MyLabelView: UILabel
+@end
+@implementation MyLabelView
+-(instancetype)init {
+    if (self = [super init]) {
+        self.textColor = UIColor.redColor;
+    }
+    return self;
+}
+@end
+
+@interface LabelBackgroundView: UIView
+@end
+@implementation LabelBackgroundView
+-(instancetype)init {
+    if (self = [super init]) {
+        self.backgroundColor = UIColor.greenColor;
+    }
+    return self;
+}
+@end
+
+@interface ViewController ()
+
+@property (nonatomic, strong, nonnull) id<LOKLayout> primaryLayout;
+@property (nonatomic, strong, nonnull) id<LOKLayout> helloWorldLayout;
+@property (nonatomic, strong, nonnull) LOKReloadableViewLayoutAdapter *adapter;
+
+@end
+
+
+@implementation ViewController
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _helloWorldLayout = [ViewController makeHelloLayout];
+    }
+    return self;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.view.backgroundColor = UIColor.whiteColor;
+
+    UICollectionView *collectionView = [[UICollectionView alloc] initWithFrame:self.view.bounds
+                                                          collectionViewLayout:[[UICollectionViewFlowLayout alloc] init]];
+    collectionView.backgroundColor = UIColor.whiteColor;
+    [self.view addSubview:collectionView];
+    self.adapter = [[LOKReloadableViewLayoutAdapter alloc] initWithCollectionView:collectionView];
+    collectionView.delegate = self.adapter;
+    collectionView.dataSource = self.adapter;
+
+//    UITableView *tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStylePlain];
+//    [self.view addSubview:tableView];
+//    self.adapter = [[LOKReloadableViewLayoutAdapter alloc] initWithTableView:tableView];
+//    tableView.delegate = self.adapter;
+//    tableView.dataSource = self.adapter;
+
+    LOKSizeLayout *cellLayout = [[LOKSizeLayout alloc] initWithMinWidth:self.view.bounds.size.width
```

Comment:
Have you considered creating a few convenience initializers?

Reply:
There are too many parameters with default properties, so making convenience initializers even for the common combinations would be too many. A builder pattern will be more appropriate and that's coming in a separate change.